### ISSUE TITLE
Mixed usage of case class and class with serializable interface.

### ISSUE THREAD

**Issue by HyukjinKwon** [apache/incubator-predictionio#203]
I see the mixed usage of `case` classes with serializable and normal classes with serializable.
Scala `case` class itself is serializable by default. I think it does not have to batter extending `Serializable` and also making them as normal classes for all `case` classes with serializable and normal classes with serializable.

For example,

There is a normal class at [RandomModel.scala#L3-L5
](https://github.com/PredictionIO/PredictionIO/blob/65a64c902ad7671dfb2fb0e29d8af806a01c55d0/examples/experimental/scala-local-friend-recommendation/src/main/scala/RandomModel.scala#L3-L5)

```scala
class RandomModel(
  val randomThreshold: Double
) extends Serializable
```

This can be just

```scala
case class RandomModel(
  val randomThreshold: Double
)
```

Another example is the class at [Engine.scala#L6-L9](https://github.com/PredictionIO/PredictionIO/blob/master/examples/experimental/scala-parallel-recommendation-custom-datasource/src/main/scala/Engine.scala#L6-L9).

```scala
case class Query(
  user: String,
  num: Int
) extends Serializable
```

`extends Serializable` can be removed as below:

```scala
case class Query(
  user: String,
  num: Int
)
```

Are they intendedly used in a mixed way?
I would like to submit a PR to simply fix them if this is really an issue.

**Comment by yipjustin** [comment]
Yes. Please submit a PR. For these cases, a plain "case class" suffices.

**Comment by HyukjinKwon** [comment]
Fixed in #205
